x-pack/legacy/plugins/maps/public/angular/map_controller.js

```diff
@@ -174,26 +174,27 @@ app.controller('GisMapController', ($scope, $route, kbnUrl, localStorage, AppSta
     }
   }
 
-  $scope.$watch('savedQuery', (newSavedQuery, oldSavedQuery) => {
+  $scope.$watch('savedQuery', (newSavedQuery) => {
     if (!newSavedQuery) return;
 
     $state.savedQuery = newSavedQuery.id;
-
-    if (newSavedQuery.id === (oldSavedQuery && oldSavedQuery.id)) {
-      updateStateFromSavedQuery(newSavedQuery);
-    }
+    updateStateFromSavedQuery(newSavedQuery);
```

Comment:
Since the `$state.savedQuery` watcher won't re-request the saved query we already have in hand, it also won't call `updateStateFromSavedQuery`. So now we can call it here without any conditional.